opengever/maintenance/scripts/reopen_and_close_rr_dossiers.py

```diff
@@ -0,0 +1,144 @@
+
+        # Wrap the user in the acquisition context of the portal
+        tmp_user = tmp_user.__of__(api.portal.get().acl_users)
+        newSecurityManager(getRequest(), tmp_user)
+
+        yield
+    finally:
+        # Restore the old security manager
+        setSecurityManager(old_manager)
+
+
+class RRDossierManager(object):
+
+    def write_list_to_the_annotiations(self, uids):
+        ann = IAnnotations(api.portal.get())
+        if ANNOTATIONS_KEY not in ann:
+            ann[ANNOTATIONS_KEY] = PersistentList(uids)
+        else:
+            uids = list(set(list(ann[ANNOTATIONS_KEY]) + uids))
+            ann[ANNOTATIONS_KEY] = PersistentList(uids)
+
+    def open_dossiers(self, start_date):
+        catalog = api.portal.get_tool('portal_catalog')
+        date_range = {'query': start_date, 'range': 'min'}
+
+        brains = catalog.unrestrictedSearchResults(
+            {'object_provides': IDossierMarker.__identifier__,
+             'review_state': 'dossier-state-resolved',
+             'start': date_range})
+
+        uids = []
+        for brain in brains:
+            dossier = brain.getObject()
+            with elevated_privileges():
+                Reactivator(dossier).reactivate()
+
+            uids.append(brain.UID)
+            logger.info('{} reopend'.format(brain.getURL()))
+
+        self.write_list_to_the_annotiations(uids)
+        logger.info('{} dossiers reopend'.format(len(uids)))
+
+    def close_dossiers(self):
+        ann = IAnnotations(api.portal.get())
+        if ANNOTATIONS_KEY not in ann:
+            raise Exception('No dossiers reopened dossiers stored - nothing to do.')
+
+        uids = ann[ANNOTATIONS_KEY]
+        for uid in uids:
+            dossier = uuidToObject(uid)
+            with elevated_privileges():
+                try:
+                    LockingResolveManager(dossier).resolve()
+                    logger.info('{} closed'.format(dossier.absolute_url()))
+                except PreconditionsViolated:
+                    logger.error('Dossier {} could not been resolved'.format(
+                        dossier.absolute_url()))
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('mode', choices=['reopen', 'close'])
+    parser.add_argument('-s', dest='site_root', default=None,
+                        help='Absolute path to the Plone site')
+    parser.add_argument('-d', dest='start_date', default=None,
+                        help='Start date in ?? format')
+    parser.add_argument('-n', dest='dryrun', default=False, help='Dryrun')
+    options = parser.parse_args(sys.argv[3:])
+    app = setup_app()
+
+    setup_plone(app, options)
+
+    if options.dryrun:
+        transaction.doom()
+
+    if options.mode == 'reopen':
+        if not options.start_date:
+            raise Exception('start_date is required in reopen mode')
+
+        start_date = datetime.strptime(options.start_date, INPUT_DATEFMT)
```

Comment:
Parsing a date input from command line seems a bit prone to errors, so giving some feedback on how it was parsed would be nice. So maybe log the start_date, or print it.